crates/subspace-node/src/lib.rs

```diff
@@ -184,6 +184,10 @@ pub struct Cli {
     #[clap(long)]
     pub dsn_bootstrap_node: Vec<Multiaddr>,
 
+    /// DSN configuration argument: record cache size in items.
+    #[clap(long, default_value_t = 32768)]
+    pub dsn_record_cache_size: usize,
```

Comment:
I think this should be called something different and used for limiting cache of local pieces rather that accepting any puts from the outside.